src/librustc_typeck/check/wfcheck.rs

```diff
@@ -646,6 +687,18 @@ fn error_380(ccx: &CrateCtxt, span: Span) {
                Trait for ..`) must have no methods or associated items")
 }
 
+fn error_565(ccx: &CrateCtxt, span: Span) {
+    span_err!(ccx.tcx.sess, span, E0565,
+              "traits with default impls (`e.g. unsafe impl \
+               Trait for ..`) can not have predicates")
+}
+
+fn error_566(ccx: &CrateCtxt, span: Span) {
```

Comment:
I don't think we need these helpers if the error is only reported from one place. Also, let's write:

"auto traits (i.e., those with `impl Trait for ..`) cannot have predicates"

Might as well start pushing the approved terminology.

Since this is a new error, even if the code is unstable, it's probably best to add a note saying "this rule is the result of a recent soundness fix; see #29859 for details".